python/ql/src/semmle/python/Files.qll

```diff
@@ -89,7 +89,10 @@ class File extends Container {
         i.getTest().(Compare).compares(name, op, main) and
         name.getId() = "__name__" and
         main.getText() = "__main__"
-      )
+      ) and
+      // Exclude files named `__main__.py`. These are often _not_ meant to be run directly, but
+      // contain this construct anyway.
```

Comment:
I think it would be nice with more of your (excellent) comment to explain why this is needed.

```suggestion
      // Exclude files named `__main__.py`. These are often _not_ meant to be run directly, but
      // contain this construct anyway.
      //
      // Their presence in a package (say, `foo`) means one can execute the package directly using
      // `python -m foo` (which will run the `foo/__main__.py` file). Since being an entry point for
      // execution means treating imports as absolute, this causes trouble, since when run with
      // `python -m`, the interpreter uses the usual package semantics.
```